api/envoy/extensions/filters/network/zookeeper_proxy/v3/zookeeper_proxy.proto

```diff
@@ -39,4 +40,60 @@ message ZooKeeperProxy {
   //
   // if that is set. If it isn't, ZooKeeper's default is also 1Mb.
   google.protobuf.UInt32Value max_packet_bytes = 3;
+
+  // List of latency thresholds for opcodes to decide the fast/slow responses and emit metrics (used for error budget calculation).
+  //
+  // https://sre.google/workbook/implementing-slos/
+  //
+  // If no latency threshold is set, the fast/slow response counters will always be 0.
+  // If the default threshold is not set, it will be set as 100 millisecond.
+  // If the threshold of one opcode is not set, it will fallback to use the default
+  // threshold.
+  //
+  // Specifying latency thresholds multiple times for one opcode is not allowed.
+  repeated LatencyThreshold latency_thresholds = 4 [(validate.rules).repeated = {max_items: 27}];
+}
+
+message LatencyThreshold {
+  enum Opcode {
+    Default = 0;
```

Comment:
Looking at the [ZooDef.java](https://github.com/apache/zookeeper/blob/89c1831f84891f425f1fa9224210587124f1c1ec/zookeeper-server/src/main/java/org/apache/zookeeper/ZooDefs.java#L38) declaration, this seems to be reserved for `notification`. I think the "default" value should be explicitly defined in its own field, and these will serve as overrides for a specific Opcode.

Reply:
We cannot make the proto value 1:1 mapping to the opcode value since some opcode values are negative. So we maintains a map in the code.

If we move the default threshold to a separate field within the `LatencyThreshold` message, I think  it will only serve as the default value of one specific opcode instead of the default value of all opcode since we can repeat the `LatencyThreshold` message.

If we move the default threshold under the `ZooKeeperProxy` message, then there are two fields related to latency threshold, will it confuse people?

Reply:
cc @adisuissa, any thoughts? Thanks

Reply:
Yeah, agree. Just moved the default latency threshold under `ZooKeeperProxy`.